server/swap/swap.go

```diff
@@ -0,0 +1,1171 @@
+	if err != nil {
+		s.respondError(*resp.ID, acker.user, rpc.SerializationError,
+			fmt.Sprintf("unable to serialize mathch params: %v", err))
+		return
+	}
+	err = s.authMgr.Auth(acker.user, msg, sigBytes)
+	if err != nil {
+		s.respondError(*resp.ID, acker.user, rpc.SignatureError,
+			fmt.Sprintf("signature validation error: %v", err))
+		return
+	}
+	// Set the appropriate signature, based on the current step and actor.
+	// DRAFT NOTE: Using the matchMtx like this is not ideal. It might be better
+	// to have individual locks on the matchTrackers.
+	s.matchMtx.Lock()
+	defer s.matchMtx.Unlock()
+	if acker.isAudit {
+		if acker.isMaker {
+			acker.match.Sigs.MakerAudit = sigBytes
+		} else {
+			acker.match.Sigs.TakerAudit = sigBytes
+		}
+	} else {
+		if acker.isMaker {
+			acker.match.Sigs.MakerRedeem = sigBytes
+		} else {
+			acker.match.Sigs.TakerRedeem = sigBytes
+		}
+	}
+}
+
+// processInit processes the `init` RPC request, which is used to inform the
+// DEX of a newly broadcast swap transaction. Once the transaction is seen and
+// and audited by the Swapper, the counter-party is informed with an 'audit'
+// request. This method is run as a chainWaiter.
+func (s *Swapper) processInit(req *rpc.Request, params *rpc.InitParams, stepInfo *stepInformation) bool {
+	if !rpc.IsValidIDType(req.ID) {
+		s.respondError(-1, stepInfo.actor.user, rpc.IDTypeError,
+			fmt.Sprintf("unsupported ID type: %T", req.ID))
+		return dontTryAgain
+	}
+	// Validate the swap contract
+	chain := stepInfo.asset.Backend
+	actor, counterParty := stepInfo.actor, stepInfo.counterParty
+	tx, err := chain.Transaction(params.TxID)
+	if err != nil {
+		// If there is an error, don't give up yet, since it could be due to network
+		// latency. Check again on the next tick.
+		// DRAFT NOTE: Could get a little smarter here by using go 1.13 (Error).Is
+		// to check that the transaction was not found, and not some other less
+		// recoverable state. Should require minimal modification of backends.
+		return tryAgain
+	}
+	// Decode the contract and audit the contract.
+	contract, err := hex.DecodeString(params.Contract)
+	if err != nil {
+		s.respondError(req.ID, actor.user, rpc.ContractError, fmt.Sprintf("error decoding contract: %v", err))
+		return dontTryAgain
+	}
+	recipient, val, err := tx.AuditContract(params.Vout, contract)
+	if err != nil {
+		s.respondError(req.ID, actor.user, rpc.ContractError, fmt.Sprintf("error auditing contract: %v", err))
+		return dontTryAgain
+	}
+	if recipient != counterParty.order.SwapAddress() {
+		s.respondError(req.ID, actor.user, rpc.ContractError,
+			fmt.Sprintf("incorrect recipient. expected %s. got %s", recipient, counterParty.order.SwapAddress()))
+		return dontTryAgain
+	}
+	if val != stepInfo.checkVal {
+		s.respondError(req.ID, actor.user, rpc.ContractError,
+			fmt.Sprintf("contract error. expected contract value to be %d, got %d", stepInfo.checkVal, val))
+		return dontTryAgain
+	}
+
```

Comment:
The `DEXAsset` likely needs a `FeeRate` method so we can check fees here. Will handle in a separate PR.